pkg/client/restconfig/restconfig.go

```diff
@@ -39,18 +39,24 @@ func NewRestConfig(timeout time.Duration) (*rest.Config, error) {
 	var cfg *rest.Config
 	// Detect kubectl config file
 	path, err := KubeConfigPath()
-	if err != nil {
-		// Build from k8s downward API
-		cfg, err = NewFromInClusterConfig()
-		if err != nil {
-			return nil, fmt.Errorf("failed to build rest config: kubeconfig not found: reading in-cluster config: %w", err)
+	kubeConfigExists := false
+	if err == nil {
+		if _, err := os.Stat(path); err == nil {
```

Comment:
I think there are other possible errors from stat. Might want to check if it's a missing file error specifically.

Reply:
This is checking for existence, the err should only be nil if the file exists.

Agreed if we were checking the inverse that we would need to check the error type.

Reply:
Ah, indeed. So used to seeing the other pattern I misread it.